### PR TITLE
Update collectors to use new Config interface

### PR DESCRIPTION
**Pull request by rfratto**
This PR updates all the collectors to use the new Config interface. 

After this PR, feat-importable should be "finished" and be usable inside grafana/agent.

### CODE REVIEW

File: collector/dfsr.go [hunk 1]
```diff
@@ -8,16 +8,22 @@ import (
...
+	EnabledCollectors string
+}
 
-func init() {
-	// Perflib sources are dynamic, depending on the enabled child collectors
-	var perflibDependencies []string
-	for _, source := range expandEnabledChildCollectors(*dfsrEnabledCollectors) {
-		perflibDependencies = append(perflibDependencies, dfsrGetPerfObjectName(source))
-	}
+// Name implemets Config.
```
Comment (mattdurham) [inline collector/dfsr.go]: typo

**Review by mattdurham — APPROVED** [review]
One small typo but LGTM

**Comment by rfratto** [comment]
Ready for review. This should be functional now.

File: collector/process.go [hunk 2]
```diff
@@ -14,10 +14,30 @@ import (
...
+	ka.Flag(
+		"collector.process.blacklist",
+		"Regexp of processes to exclude. Process name must both match whitelist and not match blacklist to be included.",
+	).Default("").StringVar(&c.ProcessBlacklist)
+}
+
+// Build implements Config.
+func (c *ProcessConfig) Build() (Collector, error) { return newProcessCollector() }
+
 var (
```
Comment (mattdurham) [inline collector/process.go]: The kingpin flags are double dipping here which causes an error

Comment (rfratto) [reply collector/process.go]: Thanks, fixed :+1:

File: collector/cache.go [hunk 3]
```diff
@@ -225,6 +234,11 @@ func newCacheCollector() (Collector, error) {
 	}, nil
 }
 
+// CacheCollector implements Collector.
```
Comment (mattdurham) [inline collector/cache.go]: nit: Should this be `PerfCounterNames implements Collector`?

**Review by mattdurham — APPROVED** [review]
one small nit

**Pull request merged**